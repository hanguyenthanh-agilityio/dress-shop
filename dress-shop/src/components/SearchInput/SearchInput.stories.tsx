import { StoryObj, Meta } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';
import SearchInput from '.';

export default {
  title: 'Components/SearchInput',
  component: SearchInput,
  decorators: [
    (Story) => (
      <MemoryRouter>
        <Story />
      </MemoryRouter>
    )
  ]
} as Meta;

type Story = StoryObj<typeof SearchInput>;

export const Default: Story = {};
