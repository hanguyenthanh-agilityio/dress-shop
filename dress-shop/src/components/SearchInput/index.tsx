import { Icon, Input, InputGroup, InputRightElement } from "@chakra-ui/react";
import {Search2Icon} from "@chakra-ui/icons"

interface SearchInputProps {
  placeholder?: string;
  onChange: () => void;
}

const SearchInput = ({
  placeholder = "Search Product...",
  onChange
}: SearchInputProps) => {

  return (
    <InputGroup>
      <Input
        placeholder={placeholder}
        onChange={onChange}
        name="search-input"
        data-testid="search-input-field"
      />
      <InputRightElement
        children={
          <Icon
            as={Search2Icon}
            color="text.helper"
            w="30px"
            h="30px"
          />
        }
      />
    </InputGroup>
  )
}

export default SearchInput;
