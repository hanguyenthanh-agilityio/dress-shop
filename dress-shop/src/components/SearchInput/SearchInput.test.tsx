import {render } from "@testing-library/react";
import "@testing-library/jest-dom";
import SearchInput from ".";


describe("SearchInput component", () => {
  const props = {
    placeholder: "Search",
    onChange: jest.fn(),
  };

  const search = () => {
    return render(<SearchInput {...props} />);
  };

  it("Should render SearchInput snapshot correctly", () => {
    expect(search()).toMatchSnapshot();
  });
});
