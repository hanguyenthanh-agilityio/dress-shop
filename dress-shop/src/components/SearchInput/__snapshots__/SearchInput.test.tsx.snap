// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component Should render SearchInput snapshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="chakra-input__group css-bx0blc"
        data-group="true"
      >
        <input
          class="chakra-input css-0"
          data-testid="search-input-field"
          name="search-input"
          placeholder="Search"
        />
        <div
          class="chakra-input__right-element css-dp9r5f"
        >
          <svg
            class="chakra-icon chakra-icon css-nu595g"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M23.414,20.591l-4.645-4.645a10.256,10.256,0,1,0-2.828,2.829l4.645,4.644a2.025,2.025,0,0,0,2.828,0A2,2,0,0,0,23.414,20.591ZM10.25,3.005A7.25,7.25,0,1,1,3,10.255,7.258,7.258,0,0,1,10.25,3.005Z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-input__group css-bx0blc"
      data-group="true"
    >
      <input
        class="chakra-input css-0"
        data-testid="search-input-field"
        name="search-input"
        placeholder="Search"
      />
      <div
        class="chakra-input__right-element css-dp9r5f"
      >
        <svg
          class="chakra-icon chakra-icon css-nu595g"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M23.414,20.591l-4.645-4.645a10.256,10.256,0,1,0-2.828,2.829l4.645,4.644a2.025,2.025,0,0,0,2.828,0A2,2,0,0,0,23.414,20.591ZM10.25,3.005A7.25,7.25,0,1,1,3,10.255,7.258,7.258,0,0,1,10.25,3.005Z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
